wordle-app/pages/WordOfTheDayGame.vue

```diff
@@ -0,0 +1,290 @@
+
+const router = useRouter();
+const game = reactive(new Game());
+game.startNewGame();
+provide("GAME", game);
+
+// User name dialog
+const showUserNameDialog = ref(false);
+const nameUserNameDialog = ref("");
+
+function nameDialogClosed() {
+  showUserNameDialog.value = false;
+  startStopwatch();
+}
+
+var showWordsList = ref(false);
+
+onMounted(() => {
+
+  window.addEventListener("keyup", onKeyup);
+
+
+  // Assignment 3 Task 2
+  // Check if user name is stored in local storage
+  checkUserNameLocalStorage();
+  checkUserName();
+
+  if(!showUserNameDialog.value){
+    startStopwatch();
+  }
+
+});
+
+
+onUnmounted(() => {
+  window.removeEventListener("keyup", onKeyup);
+});
+
+async function getWordFromApi(): Promise<string> {
+  let wordUrl = "word/wordOfTheDay";
+  const response = await Axios.get(wordUrl);
+  return response.data;
+}
+
+function onKeyup(event: KeyboardEvent) {
+  if (showUserNameDialog.value) {
+    if (event.key === 'Enter') {
+      showUserNameDialog.value = false;
+      nuxtStorage.localStorage.setData("userName", nameUserNameDialog.value);
+    }
+  } else {
+    if (event.key === "Enter") {
+      game.submitGuess();
+    } else if (event.key == "Backspace") {
+      game.removeLastLetter();
+    } else if (event.key.match(/[A-z]/) && event.key.length === 1) {
+      game.addLetter(event.key.toUpperCase());
+    }
+  }
+
+}
+
+async function checkUserNameLocalStorage() {
+  
+  if(nuxtStorage.localStorage.getData("userName") != null){
+    nameUserNameDialog.value = nuxtStorage.localStorage.getData("userName");
+    showUserNameDialog.value = false;
+  }
+  else {
+    nameUserNameDialog.value = "Guest";
+    showUserNameDialog.value = true;
+  }
+
+}
+
+async function checkUserName() {
+  const response = await Axios.get("/Player/Player?playerName=" + nameUserNameDialog.value);
+  const player: Player = response.data;
+
+  if(player.name == "Guest"){
+    showUserNameDialog.value = true;
+  }
+}
+
+function postScore() {
+    console.log("PostScoreEntered")
+    //console.log("Stopwatch seconds: " + stopwatch.seconds.value)
+    var attempts = 0;
+    if(game.gameState == GameState.Won){
+      attempts = game.guessIndex + 1;
+    }else{
+      attempts = game.guesses.length + 5;
+    }
+    Axios.post("Player/AddPlayer", {
+      Name: nameUserNameDialog.value,
+      GameCount: 1,
+      AverageAttempts: attempts,
+      AverageSecondsPerGame: seconds.value,
+    })
+    .then(res => {
+      console.log(res.data);
+    })
+    .catch(err => {
+      console.log("Error" + err);
+    })
+}
+
+function updateWordDate() {
+    console.log("UpdateWordDateEntered")
+    var attempts = 0;
+    if(game.gameState == GameState.Won){
+      attempts = game.guessIndex + 1;
+    }else{
+      attempts = game.guesses.length + 5;
+    }
+    var nameList = new Array(nameUserNameDialog.value);
+    const today = new Date();
+    var curDate = today.getMonth() + "/" + today.getDay() + "/" + today.getFullYear();
```

Comment:
```suggestion
    var curDate = new Date().toLocaleDateString('en-CA');
```
This will get the date within a format of yyyy-mm-dd. I think this will be better for using Datetime format if you choose to switch to Datetime format